src/crypto/CHIPCryptoPAL.h

```diff
@@ -900,6 +900,13 @@ class Hash_SHA256_stream
      */
     CHIP_ERROR Begin();
 
+    /**
+     * @brief check if the digest computation has been initialized.
+     *
+     * @return True if the context is correctly initialized; otherwise, false.
+     */
+    bool IsInitialized();
```

Comment:
Doesn't this need implementations for the non-OpenSSL/BoringSSL backends?

Reply:
And also: this looks like an API that API consumers should call before they do anything with the object.... but they very much don't do that, and if they tried it would not work with most of the backends.  Why is this public API at all?

Reply:
+1. The method used here should be used more widely whenever the context is used at all. I would expect this API to be private since it's only used internally and should never be false. The idea is to avoid double-free or leaks, right?

Reply:
Yes this was a mistake, the `IsInitialized` is only to be used internally by `Hash_SHA256_stream` within the OpenSSL backend. So I will make `IsInitialized` protected. 

@bzbarsky-apple  Should I add a stub for  `IsInitialized` in other backends or only add a comment in the `*.h` that it is only implemented in OpenSSL?

Reply:
If it's private/protected, it really does not matter; someone who tries to use it will know what they are doing and notice they need to implement it.  So comment saying "implement this if your backend needs it" is fine.